lib/settings/wifi_screen.dart

```diff
@@ -48,67 +57,122 @@ class _WifiScreenState extends State<WifiScreen> {
-      60: Icon(Icons.network_wifi_3_bar_rounded, color: Colors.green[300]),
-      40: Icon(Icons.network_wifi_2_bar_rounded, color: Colors.orange[300]),
-      20: Icon(Icons.network_wifi_1_bar_rounded, color: Colors.orange[300]),
-      0: Icon(Icons.warning_rounded, color: Colors.red[300]),
+    // Define the icons for each platform
+    final Map<int, Icon> linuxIcons = {
+      100: Icon(Icons.network_wifi_rounded, color: Colors.green[300], size: 30),
+      80: Icon(Icons.network_wifi_rounded, color: Colors.green[300], size: 30),
+      60: Icon(Icons.network_wifi_3_bar_rounded,
+          color: Colors.green[300], size: 30),
+      40: Icon(Icons.network_wifi_2_bar_rounded,
+          color: Colors.orange[300], size: 30),
+      20: Icon(Icons.network_wifi_1_bar_rounded,
+          color: Colors.orange[300], size: 30),
+      0: Icon(Icons.warning_rounded, color: Colors.red[300], size: 30),
     };
 
-    if (platform == 'linux') {
-      for (var threshold in icons.keys.toList().reversed) {
-        if (signalStrength >= threshold) {
-          return icons[threshold]!;
-        }
-      }
-    } else {
-      icons = {
-        -0: Icon(Icons.network_wifi_rounded, color: Colors.green[300]),
-        -50: Icon(Icons.network_wifi_3_bar_rounded, color: Colors.green[300]),
-        -70: Icon(Icons.network_wifi_2_bar_rounded, color: Colors.orange[300]),
-        -90: Icon(Icons.warning_rounded, color: Colors.red[300]),
-      };
+    final Map<int, Icon> otherIcons = {
+      10: Icon(Icons.network_wifi_rounded, color: Colors.green[300], size: 30),
+      0: Icon(Icons.network_wifi_rounded, color: Colors.green[300], size: 30),
+      -50: Icon(Icons.network_wifi_3_bar_rounded,
+          color: Colors.green[300], size: 30),
+      -70: Icon(Icons.network_wifi_2_bar_rounded,
+          color: Colors.orange[300], size: 30),
+      -90: Icon(Icons.warning_rounded, color: Colors.red[300], size: 30),
+    };
 
-      signalStrength = signalStrength.abs();
-      for (var threshold in icons.keys.toList().reversed) {
-        if (signalStrength >= threshold.abs()) {
-          return icons[threshold]!;
-        }
+    // Choose the correct map of icons based on the platform
+    final Map<int, Icon> icons = platform == 'linux' ? linuxIcons : otherIcons;
+
+    // Find the first icon where the signal strength is less than or equal to the threshold
+    for (var threshold in icons.keys.toList().reversed) {
+      if (signalStrength <= threshold) {
+        return icons[threshold]!;
       }
     }
+
+    // If no icon was found, return a warning icon
     return const Icon(Icons.warning_rounded);
   }
 
   late String platform;
-  Future<List<Map<String, String>>> _getWifiNetworks() async {
+
+  Future<List<Map<String, String>>> _getWifiNetworks(
+      {bool alreadyConnected = false}) async {
     wifiNetworks.clear();
     try {
+      /*if (Theme.of(context).platform == TargetPlatform.macOS &&
+          !alreadyConnected) {
```

Comment:
Should we remove this as well?

Reply:
Resolved by [17dd19d](https://github.com/TheContrappostoShop/Orion/pull/14/commits/17dd19d100d306003192341c4a4ca3d23aad545d)